tests/control/PlanarArmControllerTest.cpp

```diff
@@ -1,13 +1,60 @@
 #include "../../src/control/PlanarArmController.h"
 
+#include "../../src/Constants.h"
+
+#include <iostream>
+
 #include <catch2/catch.hpp>
 
 using namespace Catch::literals;
 using namespace control;
 using namespace std::chrono_literals;
 
+constexpr double PI = M_PI;
+
+std::string toString(const Eigen::Vector2d& pose) {
+	std::stringstream ss;
+	ss << "(" << pose(0) << ", " << pose(1) << ")";
+	return ss.str();
+}
+
+void assertApprox(const Eigen::Vector2d& p1, const Eigen::Vector2d& p2, double dist = 1e-5,
+				  double angle = 1e-5) {
+	Eigen::Vector2d diff = p1 - p2;
+	bool distEqual = diff.norm() <= dist;
+	double thetaDiff = std::fmod(abs(diff(1)), 2 * PI);
+	// change domain from [0, 2pi) to (-pi, pi]
+	if (thetaDiff > PI) {
+		thetaDiff -= 2 * PI;
+	}
+	bool angleEqual = abs(thetaDiff) <= angle;
+	if (distEqual && angleEqual) {
+		SUCCEED();
+	} else {
+		std::cout << "Expected: " << toString(p1) << ", Actual: " << toString(p2) << std::endl;
+		FAIL();
+	}
```

Comment:
Instead of this, we should use `INFO` along with `REQUIRE`, which has all the functionality you need, and works simpler. So this could be replaced with:

```c++
std::stringstream ss;
ss << "Expected: " << toString(p1) << ", Actual: " << toString(p2);
INFO(ss.str());
REQUIRE(distEqual && angleEqual);
```

More info is [here](https://github.com/catchorg/Catch2/blob/devel/docs/logging.md), but note that the message given to `INFO` is only printed if a test fails, which is what we want.